src/include/platform/BuildTime.h

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+// Adapted from https://stackoverflow.com/questions/11697820/how-to-use-date-and-time-predefined-macros-in-as-two-integers-then-stri
+
+#pragma once
+
+#include <platform/CHIPDeviceConfig.h>
+
+// Expects input as provided by __DATE__ and __TIME__ macros.
+//
+// Example of __DATE__ string: "Jul 27 2012"
+// Example of __TIME__ string: "21:06:19"
+
+#define COMPUTE_BUILD_YEAR(_date)                                                                                                  \
+    (((_date)[7] - '0') * 1000 + ((_date)[8] - '0') * 100 + ((_date)[9] - '0') * 10 + ((_date)[10] - '0'))
+
+#define COMPUTE_BUILD_DAY(_date) ((((_date)[4] >= '0') ? ((_date)[4] - '0') * 10 : 0) + ((_date)[5] - '0'))
+
+#define BUILD_MONTH_IS_JAN(_date) ((_date)[0] == 'J' && (_date)[1] == 'a' && (_date)[2] == 'n')
```

Comment:
Why is the check for 'n' needed?  Starting with "Ja" is enough to uniquely identify January.